plugin-server/src/utils/db/kafka-producer-wrapper.ts

```diff
@@ -48,17 +48,21 @@ export class KafkaProducerWrapper {
     }
 
     async queueMessage(kafkaMessage: ProducerRecord): Promise<void> {
-        const messageSize = this.getMessageSize(kafkaMessage)
+        const messageSize = this.estimateMessageSize(kafkaMessage)
 
-        if (this.currentBatchSize + messageSize > this.maxBatchSize) {
+        if (this.currentBatch.length > 0 && this.currentBatchSize + messageSize > this.maxBatchSize) {
             // :TRICKY: We want to first flush then immediately add the message to the queue. Awaiting and then pushing would result in a race condition.
             await this.flush(kafkaMessage)
         } else {
             this.currentBatch.push(kafkaMessage)
             this.currentBatchSize += messageSize
 
             const timeSinceLastFlush = Date.now() - this.lastFlushTime
-            if (timeSinceLastFlush > this.flushFrequencyMs || this.currentBatch.length >= this.maxQueueSize) {
+            if (
+                this.currentBatchSize > this.maxBatchSize ||
```

Comment:
> Fix: immediately flush if enqueueing a too-large message. This way unrelated messages don't end up in DLQ and sentry reports have proper context.

Does this help here because `this.flush()` threw before (line 55)  making us miss the next message, if it happened the next time we called `queueMessage`?

Reply:
No. Previously if the batch was empty and a single too-large message came in, we would call flush() before appending and then append. Then when the next message comes along, that flush fails since the message in the queue is too large.

Now we append and flush immediately. Check the new test added - it would have failed before.

Reply:
ok, so that's what I thought - clearly I didn't word the question well, thanks for confirming.